src/program-editors.rst

```diff
@@ -39,4 +39,41 @@ TODO: work on version control systems
+A variety of tools have been developed to improve upon the accessibility of programming languages and development environments in general. One example is CodeTalk, a programming environment that attempts to extend the benefits of fully-featured IDEs to visually impaired developers :cite:`codetalk`. CodeTalk offers a variety of features focusing on 3 areas: glanceability, navigability, and debugging. Features include quick ways to get context of the current cursor location, and auditory cues while debugging. CodeTalk was evaluated through a qualitative exploratory study and survey of six participants, and is currently available as an extension for Visual Studio. APL (Auditory Programming Language) is a programming language designed specifically for blind developers that uses both sound icons and text. Preliminary findings indicate that rich audio environments may better support blind developers mental models :cite:`sanchez2005blind`.
+ 
+Additionally, prior research has explored task specific tools. StructJumper is an extension for Eclipse that creates a tree structure from code to help blind programmers navigate code :cite:`baker2015structjumper`. StructJumper was evaluated by 7 blind developers. Participants were asked to complete a series of tasks with and without the extension, and found that StructJumper allowed participants to complete tasks faster than without. Potluri et al. consider tools that could assist with the task of UI design :cite:`potluri2019ai`. The authors describe existing methods that blind and low-vision creators use to get a sense of UI aesthetics, including physical prototyping, and discuss the potential use of AI for assistance in this task, especially for higher-level features, such as color and general ‘look’.
+ 
+Programming Education for Students with Visual Impairments
+----------------------------
+As previously mentioned, programming education often uses diagrams to represent abstract concepts, a barrier for visually impaired students. Stefik et al. present an educational infrastructure for blind and visually impaired middle and high school students that consists of Sodbeans, an auditory programming environment, Hop, a programming language, and a multi-sensory curriculum :cite:`stefik2011design`. They conducted an empirical study at a programming summer camp with 12 blind participants to evaluate their framework. Similarly, Blocks4All is a block-based, touchscreen programming environment :cite:`milne2018blocks4all`. The authors identify accessibility barriers, as well as appropriate touchscreen interactions.
+ 
+Motor Impairments
+----------------------------
+Motor impairments also require input devices other than the traditional keyboard and mouse, which can be painful or impossible to use for some :cite:`saphra_2019`. Alternatives can include speech based interaction. For example, Begel and Graham evaluate a speech-based programming system called SPEED (SPEech EDitor) :cite:`begel2006assessment`.
```

Comment:
You can cite the Saphra blog post as a paper. Please describe it in more detail.

Reply:
Please also describe the Begel 2006 paper in more detail.